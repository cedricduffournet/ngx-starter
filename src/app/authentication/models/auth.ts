export interface AuthToken {
  access_token: string;
  expires_in: number;
  token_type: string;
  refresh_token: string;
}

export interface Credentials {
  login: string;
  password: string;
}

export interface ResetPassword {
  plainPassword: {
    first: string;
    second: string;
  };
}

export interface Register extends ResetPassword {
  civility: number;
  firstname: string;
  lastname: string;
  email: string;
  confirmationUrl: string;
}

export interface SendResetPassword {
  username: string;
}
