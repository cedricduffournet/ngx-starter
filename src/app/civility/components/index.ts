export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
