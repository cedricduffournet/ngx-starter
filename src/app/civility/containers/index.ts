export * from './civility-add-modal/civility-add-modal.component';
export * from './civility-update-modal/civility-update-modal.component';
export * from './civility-delete-modal/civility-delete-modal.component';
export * from './civility-list-view/civility-list-view.component';
