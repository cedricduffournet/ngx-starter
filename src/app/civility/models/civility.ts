import { schema } from 'normalizr';

export interface Civility {
  id: number;
  name: string;
  code: string;
}

export const civilitySchema = new schema.Entity('civilities');
