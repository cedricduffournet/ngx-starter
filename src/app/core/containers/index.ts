export * from './toaster';
export * from './http-progress-bar/http-progress-bar.component';
