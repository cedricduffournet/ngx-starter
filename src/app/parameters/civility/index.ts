export * from './parameter-civility.component';
