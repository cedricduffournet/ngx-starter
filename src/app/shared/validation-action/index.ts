export * from './validation-action.module';
