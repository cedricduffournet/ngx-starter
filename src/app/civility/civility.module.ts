import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityEffects } from '@app/civility/state/effects';
import {
  CivilityAddComponent,
  CivilityUpdateComponent,
  CivilityDeleteComponent,
  CivilityFormComponent,
  CivilityItemComponent,
  CivilityItemsComponent
} from '@app/civility/components';
import {
  CivilityAddModalComponent,
  CivilityUpdateModalComponent,
  CivilityDeleteModalComponent,
  CivilityListViewComponent
} from '@app/civility/containers';
import { reducers } from '@app/civility/state/reducers';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    EffectsModule.forFeature([CivilityEffects]),
    StoreModule.forFeature('civilities', reducers)
  ],
  declarations: [
    CivilityListViewComponent,
    CivilityAddComponent,
    CivilityUpdateComponent,
    CivilityDeleteComponent,
    CivilityFormComponent,
    CivilityItemComponent,
    CivilityItemsComponent,
    CivilityAddModalComponent,
    CivilityUpdateModalComponent,
    CivilityDeleteModalComponent
  ],
  entryComponents: [
    CivilityUpdateModalComponent,
    CivilityAddModalComponent,
    CivilityDeleteModalComponent
  ],
  exports: [CivilityListViewComponent]
})
export class CivilityModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
