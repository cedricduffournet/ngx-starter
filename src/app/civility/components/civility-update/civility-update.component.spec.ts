import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { TranslateModule } from '@ngx-translate/core';

import { FormModule } from '@app/shared/form';
import { ValidationActionModule } from '@app/shared/validation-action';
import {
  CivilityUpdateComponent,
  CivilityFormComponent
} from '@app/civility/components';
import { Civility } from '@app/civility/models/civility';

describe('CivilityUpdate', () => {
  let fixture: ComponentFixture<CivilityUpdateComponent>;
  let component: CivilityUpdateComponent;
  const civility: Civility = {
    id: 1,
    code: 'testcode1',
    name: 'testname1'
  };

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        TranslateModule.forRoot(),
        FormModule,
        ValidationActionModule
      ],
      declarations: [CivilityUpdateComponent, CivilityFormComponent]
    });

    fixture = TestBed.createComponent(CivilityUpdateComponent);
    component = fixture.componentInstance;
    component.civility = civility;
  });

  it('should be created', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit save event', () => {
    spyOn(component.update, 'emit');
    component.onSave({ code: 'codeupdated', name: 'nameupdated' } as Civility);
    expect(component.update.emit).toHaveBeenCalledWith({
      id: civility.id,
      civility: { code: 'codeupdated', name: 'nameupdated' }
    });
  });

  it('should indicate civility is updating', () => {
    component.updating = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit cancel event', () => {
    spyOn(component.cancel, 'emit');
    component.onCancel();
    expect(component.cancel.emit).toHaveBeenCalledWith('cancel');
  });
});
