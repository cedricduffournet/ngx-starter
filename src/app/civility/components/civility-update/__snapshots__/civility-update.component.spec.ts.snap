// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CivilityUpdate should be created 1`] = `
<app-civility-update
  cancel={[Function EventEmitter]}
  civility={[Function Object]}
  update={[Function EventEmitter]}
  updating="false"
>
  <app-civility-form
    ng-reflect-civility="[object Object]"
    ng-reflect-processing="false"
  >
    <form
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <section
        class="bg-white"
      >
        <div
          class="p-3"
        >
          <app-text-input
            class="ng-untouched ng-pristine ng-valid"
            formcontrolname="code"
            ng-reflect-label="ABBREVIATION"
            ng-reflect-name="code"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                ABBREVIATION
              </label>
              <input
                class="form-control"
                placeholder=""
                type="text"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
          <app-text-input
            class="ng-untouched ng-pristine ng-valid"
            formcontrolname="name"
            ng-reflect-label="NAME"
            ng-reflect-name="name"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                NAME
              </label>
              <input
                class="form-control"
                placeholder=""
                type="text"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
        </div>
      </section>
      <app-wrapper-validation-action>
        <div
          class="d-flex justify-content-end align-items-center p-3"
        >
          <app-validation-button
            ng-reflect-confirm-disabled="false"
            ng-reflect-label-cancel="CANCEL"
            ng-reflect-label-confirm="SAVE"
          >
            <div
              class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
              ng-reflect-klass="button-validation-action text-"
              ng-reflect-ng-class=""
            >
              
              <button
                class="btn btn-cancel btn-lg btn-light"
                ng-reflect-klass="btn btn-cancel btn-lg"
                ng-reflect-ng-class="btn-light"
                type="button"
              >
                 CANCEL 
              </button>
              
              <button
                class="btn btn-confirm btn-lg btn-primary"
                ng-reflect-klass="btn btn-confirm btn-lg"
                ng-reflect-ng-class="btn-primary"
                type="submit"
              >
                
                
                
                <span>
                  SAVE
                </span>
              </button>
            </div>
          </app-validation-button>
        </div>
      </app-wrapper-validation-action>
    </form>
  </app-civility-form>
</app-civility-update>
`;

exports[`CivilityUpdate should indicate civility is updating 1`] = `
<app-civility-update
  cancel={[Function EventEmitter]}
  civility={[Function Object]}
  update={[Function EventEmitter]}
  updating={[Function Boolean]}
>
  <app-civility-form
    ng-reflect-civility="[object Object]"
    ng-reflect-processing="true"
  >
    <form
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <section
        class="bg-white"
      >
        <div
          class="p-3"
        >
          <app-text-input
            class="ng-untouched ng-pristine ng-valid"
            formcontrolname="code"
            ng-reflect-label="ABBREVIATION"
            ng-reflect-name="code"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                ABBREVIATION
              </label>
              <input
                class="form-control"
                placeholder=""
                type="text"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
          <app-text-input
            class="ng-untouched ng-pristine ng-valid"
            formcontrolname="name"
            ng-reflect-label="NAME"
            ng-reflect-name="name"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                NAME
              </label>
              <input
                class="form-control"
                placeholder=""
                type="text"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
        </div>
      </section>
      <app-wrapper-validation-action>
        <div
          class="d-flex justify-content-end align-items-center p-3"
        >
          <app-validation-button
            ng-reflect-confirm-disabled="true"
            ng-reflect-label-cancel="CANCEL"
            ng-reflect-label-confirm="SAVE"
          >
            <div
              class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
              ng-reflect-klass="button-validation-action text-"
              ng-reflect-ng-class=""
            >
              
              
              <button
                class="btn btn-confirm btn-lg btn-primary"
                disabled=""
                ng-reflect-klass="btn btn-confirm btn-lg"
                ng-reflect-ng-class="btn-primary"
                type="submit"
              >
                
                <span
                  class="spinner-border spinner-border-sm align-baseline"
                  style="font-size: 1rem;"
                />
                
                <span>
                  Â LOADING
                </span>
                
              </button>
            </div>
          </app-validation-button>
        </div>
      </app-wrapper-validation-action>
    </form>
  </app-civility-form>
</app-civility-update>
`;
